### PR TITLE
treewide: freebsd13 -> freebsd

### PR DESCRIPTION
**Pull request by Ericson2314**
## Description of changes

@rhelmot and @artemist think this is correct. @alyssais was once quite convinced it was *not* correct. I have no idea, but we should decide soon so we can all be happy for 24.05.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rhelmot** [comment]
Some references in favor of dropping the version number

https://wiki.freebsd.org/BinaryCompatibility
https://wiki.freebsd.org/KernelModulePackages (just the first paragraph)
https://github.com/NixOS/nixpkgs/pull/254801#issuecomment-1984834380

tl;dr FreeBSD provides syscall abi backwards compatibility. We can convert that to forwards compatibility by running our buildbots on the oldest supported (wrt nixpkgs) version of FreeBSD. This can be further converted to forwards compatibility for _everyone_ once the [jail code](https://github.com/NixOS/nix/pull/9968) is merged, nix can simply be configured to build in a jail claiming to be the lowest supported version.

**Comment by Aleksanaa** [comment]
> tl;dr FreeBSD provides syscall abi backwards compatibility. We can convert that to forwards compatibility by running our buildbots on the oldest supported (wrt nixpkgs) version of FreeBSD. This can be further converted to forwards compatibility for everyone once the https://github.com/NixOS/nix/pull/9968 is merged, nix can simply be configured to build in a jail claiming to be the lowest supported version.

Thanks for pointing it out! When NixBSD chose to use freebsd (instead of netbsd), and removed the version number from the code (similar to this PR), I once doubted the correctness of what they did.

**Comment by artemist** [comment]
The decision to use FreeBSD and to remove the version number were very practical.

Removing the version number removes a lot of manual work adding new platforms, and it's not even that useful: if you're worried about forward compatibility than a binary built for 13.2-RELEASE isn't guaranteed to run on 13.0-RELEASE.

NetBSD can't easily support sandboxed builders, so if we wanted pure native builds we'd have to figure out how to do something weird (maybe boot a vm for each derivation off virtio-9p of that's possible?). FreeBSD makes this relatively easy with jails.

**Comment by Ericson2314** [comment]
So the ABI can grow but not changing in a breaking away from 13.0 -> 13.2 and also from 13.03 -> 14? Are the ABI guarantees more or less the same regardless of whether it a difference in the major or minor number?

File: lib/systems/parse.nix [hunk 1]
```diff
@@ -326,11 +326,7 @@ rec {
     # the normalized name for macOS.
     macos    = { execFormat = macho;   families = { inherit darwin; }; name = "darwin"; };
     ios      = { execFormat = macho;   families = { inherit darwin; }; };
-    # A tricky thing about FreeBSD is that there is no stable ABI across
-    # versions. That means that putting in the version as part of the
-    # config string is paramount.
-    freebsd12 = { execFormat = elf;     families = { inherit bsd; }; name = "freebsd"; version = 12; };
-    freebsd13 = { execFormat = elf;     families = { inherit bsd; }; name = "freebsd"; version = 13; };
+    freebsd = { execFormat = elf;     families = { inherit bsd; }; name = "freebsd"; };
```
Comment (alyssais) [inline lib/systems/parse.nix:329]:
```suggestion
    freebsd  = { execFormat = elf;     families = { inherit bsd; }; name = "freebsd"; };
```

**Review by alyssais — CHANGES_REQUESTED** [review]
Makes sense to me.

**Comment by Ericson2314** [comment]
@alyssais that's an approval, with the indentation fixed?

**Review by alyssais — APPROVED** [review]
